web/src/components/molecules/Asset/Asset/AssetBody/unzipFileList.tsx

```diff
@@ -1,76 +1,42 @@
 import { Key } from "rc-table/lib/interface";
-import { CSSProperties, useState } from "react";
+import { CSSProperties, useCallback, useEffect, useState } from "react";
 
 import Icon from "@reearth-cms/components/atoms/Icon";
 import Tree, { DataNode, TreeProps } from "@reearth-cms/components/atoms/Tree";
+import { AssetFile } from "@reearth-cms/components/molecules/Asset/asset.type";
 
 type Props = {
+  file: AssetFile;
   style?: CSSProperties;
 };
 
-// TODO: this data is hard-codded, actual data should be formatted in a similar way.
-const treeData: DataNode[] = [
-  {
-    title: "Folder",
-    key: "0-0",
-    children: [
-      {
-        title: "Folder",
-        key: "0-0-0",
-        children: [
-          {
-            title: "File",
-            key: "0-0-0-0",
-          },
-          {
-            title: "File",
-            key: "0-0-0-1",
-          },
-          {
-            title: "File",
-            key: "0-0-0-2",
-          },
-        ],
-      },
-      {
-        title: "Folder",
-        key: "0-0-1",
-        children: [
-          {
-            title: "File",
-            key: "0-0-1-0",
-          },
-        ],
-      },
-      {
-        title: "Folder",
-        key: "0-0-2",
-        children: [
-          {
-            title: "File",
-            key: "0-0-2-0",
-          },
-          {
-            title: "File",
-            key: "0-0-2-1",
-          },
-        ],
-      },
-    ],
-  },
-  {
-    title: "File",
-    key: "0-1",
-  },
-  {
-    title: "File",
-    key: "0-2",
-  },
-];
-
-const UnzipFileList: React.FC<Props> = ({ style }) => {
-  const [expandedKeys, setExpandedKeys] = useState<DataNode["key"][]>(["0-0"]);
+const UnzipFileList: React.FC<Props> = ({ file, style }) => {
+  const [expandedKeys, setExpandedKeys] = useState<DataNode["key"][]>(["0"]);
   const [selectedKeys, setSelectedKeys] = useState<DataNode["key"][]>([]);
+  const [treeData, setTreeData] = useState<DataNode[]>([]);
+
+  const getTreeData = useCallback((assetFile: AssetFile, key = ""): DataNode[] => {
```

Comment:
Is this used anywhere else where passing the key is necessary? Can you not just rely on the index to keep things simpler?

Reply:
Can you please read the code again, this is a recursive function which is called from inside itself.
Currently we support 5 levels of nested children of files. and each child of each level should have a unique key.
something like this:
```ts
const treeData: DataNode[] = [
  {
    title: "Folder",
    key: "0",
    children: [
      {
        title: "Folder",
        key: "0-0",
        children: [
          {
            title: "File",
            key: "0-0-0",
          },
          {
            title: "File",
            key: "0-0-1",
          },
          {
            title: "File",
            key: "0-0-2",
          },
        ],
      },
    ],
  },
];
```

Reply:
Gotcha. Okay, I think your logic is a little off. How does this look?

```
 const getTreeData = useCallback(
    (assetFile: AssetFile, key?: string): DataNode[] =>
      assetFile?.children?.map((file: AssetFile, index: number) => {
        let children: DataNode[] = [];
        const parentKey = key ? key + "-" + index.toString() :  index.toString();

        if (file.children && file.children.length > 0) {
          children = getTreeData(
            file,
            key: parentKey + "-" + index.toString(),
          );
        }

        return {
          title: file.name,
          key: parentKey,
          children: children,
        };
      }) || [],
    [],
  );
```

Reply:
I think that this is incorrect, do you mean 
```ts
  const getTreeData = useCallback(
    (assetFile: AssetFile, key?: string): DataNode[] =>
      assetFile?.children?.map((file: AssetFile, index: number) => {
        let children: DataNode[] = [];
        const parentKey = key ? key + "-" + index.toString() : index.toString();

        if (file.children && file.children.length > 0) {
          children = getTreeData(file, parentKey);
        }

        return {
          title: file.name,
          key: parentKey,
          children: children,
        };
      }) || [],
    [],
  );
```

Reply:
Exactly. The logic is correct, right?

Reply:
Right.

Reply:
Done!